crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1294,6 +1295,27 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
         payment_data.payment_intent = payment_intent;
         payment_data.payment_attempt = payment_attempt;
 
+        let client_src = header_payload
+            .client_source
+            .clone()
+            .or(payment_data.payment_attempt.client_source.clone());
+
+        let client_ver = header_payload
+            .client_version
+            .clone()
+            .or(payment_data.payment_attempt.client_version.clone());
+
+        let frm_message = payment_data.frm_message.clone();
+
+        _req_state
```

Comment:
```suggestion
        req_state
```

Reply:
_req_state is the only variable coming from above 
` _req_state: ReqState,`